src/sentry/templates/sentry/js-sdk-loader.ts

```diff
@@ -71,7 +70,39 @@ declare const __LOADER__IS_LAZY__: any;
     });
   }
 
-  function injectSdk(callbacks) {
+  function onSentryScriptLoaded() {
+    try {
+      _window.removeEventListener(_errorEvent, onError);
+      _window.removeEventListener(_unhandledrejectionEvent, onUnhandledRejection);
+
+      // Add loader as SDK source
+      _window.SENTRY_SDK_SOURCE = 'loader';
+
+      const SDK = _window[_namespace];
+
+      const oldInit = SDK.init;
+
+      // Configure it using provided DSN and config object
+      SDK.init = function (options) {
+        const target = _config;
+        for (const key in options) {
+          if (Object.prototype.hasOwnProperty.call(options, key)) {
+            target[key] = options[key];
+          }
+        }
+
+        setupDefaultIntegrations(target, SDK);
+        oldInit(target);
+      };
+
+      // Wait a tick to ensure that all `Sentry.onLoad()` callbacks have been registered
+      setTimeout(() => setupSDK(SDK));
```

Comment:
I extracted all of this into a function & adjusted it so that we only call `setupSDK` in the next tick, which hopefully makes all of this a bit clearer!